### PR TITLE
chore: change inserts type assertions and error suppressions

### PR DESCRIPTION
**Pull request by abhiomkar**
Project import generated by Copybara.

### CODE REVIEW

**Comment by mprobst** [comment]
LGTM from me (and thanks for upstreaming @abhiomkar !).

File: packages/mdc-switch/component.ts [hunk 1]
```diff
@@ -96,23 +101,30 @@ export class MDCSwitch extends MDCComponent<MDCSwitchFoundation> implements MDCR
...
       addClass: (className: string) => rippleSurface.classList.add(className),
       computeBoundingRect: () => rippleSurface.getBoundingClientRect(),
-      deregisterInteractionHandler: (evtType: EventType, handler: EventListener) => {
-        this.nativeControl_.removeEventListener(evtType, handler, applyPassive());
-      },
+      deregisterInteractionHandler:
+          ((evtType: EventType, handler: EventListener) => {
+            this.nativeControl_.removeEventListener(
+                evtType, handler, applyPassive());
+          }) as any,
```
Comment (asyncLiz) [inline packages/mdc-switch/component.ts]: What's the purpose of `as any` for the interaction handlers?

Comment (abhiomkar) [reply packages/mdc-switch/component.ts]: /cc @mprobst

Comment (mprobst) [reply packages/mdc-switch/component.ts]: This code fails to compile with TypeScript's `nostrictGenericChecks: false` flag (i.e. what you get from `tsc --strict` by default).

I would have left the error message as a TODO, but it's a bit bulky:

```
material_components_web/switch/component.ts:107:7 - error TS2322: Type '(evtType: "input" | "progress" | "select" | "change" | "abort" | "animationcancel" | "animationend" | "animationiteration" | "animationstart" | "auxclick" | "blur" | "cancel" | "canplay" | "canplaythrough" | "click" | "close" | "contextmenu" | "cuechange" | "dblclick" | "drag" | "dragend" | "dragenter" | "dragexit" | "dragleave" | "dragover" | "dragstart" | "drop" | "durationchange" | "emptied" | "ended" | "error" | "focus" | "focusin" | "focusout" | "gotpointercapture" | "invalid" | "keydown" | "keypress" | "keyup" | "load" | "loadeddata" | "loadedmetadata" | "loadend" | "loadstart" | "lostpointercapture" | "mousedown" | "mouseenter" | "mouseleave" | "mousemove" | "mouseout" | "mouseover" | "mouseup" | "pause" | "play" | "playing" | "pointercancel" | "pointerdown" | "pointerenter" | "pointerleave" | "pointermove" | "pointerout" | "pointerover" | "pointerup" | "ratechange" | "reset" | "resize" | "scroll" | "securitypolicyviolation" | "seeked" | "seeking" | "selectionchange" | "selectstart" | "stalled" | "submit" | "suspend" | "timeupdate" | "toggle" | "touchcancel" | "touchend" | "touchmove" | "touchstart" | "transitioncancel" | "transitionend" | "transitionrun" | "transitionstart" | "volumechange" | "waiting" | "wheel", handler: EventListener) => void' is not assignable to type '<K extends "input" | "progress" | "select" | "change" | "abort" | "animationcancel" | "animationend" | "animationiteration" | "animationstart" | "auxclick" | "blur" | "cancel" | "canplay" | "canplaythrough" | "click" | "close" | "contextmenu" | "cuechange" | "dblclick" | "drag" | "dragend" | "dragenter" | "dragexit" | "dragleave" | "dragover" | "dragstart" | "drop" | "durationchange" | "emptied" | "ended" | "error" | "focus" | "focusin" | "focusout" | "gotpointercapture" | "invalid" | "keydown" | "keypress" | "keyup" | "load" | "loadeddata" | "loadedmetadata" | "loadend" | "loadstart" | "lostpointercapture" | "mousedown" | "mouseenter" | "mouseleave" | "mousemove" | "mouseout" | "mouseover" | "mouseup" | "pause" | "play" | "playing" | "pointercancel" | "pointerdown" | "pointerenter" | "pointerleave" | "pointermove" | "pointerout" | "pointerover" | "pointerup" | "ratechange" | "reset" | "resize" | "scroll" | "securitypolicyviolation" | "seeked" | "seeking" | "selectionchange" | "selectstart" | "stalled" | "submit" | "suspend" | "timeupdate" | "toggle" | "touchcancel" | "touchend" | "touchmove" | "touchstart" | "transitioncancel" | "transitionend" | "transitionrun" | "transitionstart" | "volumechange" | "waiting" | "wheel">(evtType: K, handler: SpecificEventListener<K>) => void'.
  Types of parameters 'handler' and 'handler' are incompatible.
    Types of parameters 'evt' and 'evt' are incompatible.
      Type 'Event' is not assignable to type 'GlobalEventHandlersEventMap[K]'.
        Type 'Event' is not assignable to type 'Event & ProgressEvent<EventTarget> & UIEvent & AnimationEvent & MouseEvent & FocusEvent & DragEvent & ErrorEvent & PointerEvent & KeyboardEvent & SecurityPolicyViolationEvent & TouchEvent & TransitionEvent & WheelEvent'.
          Type 'Event' is missing the following properties from type 'ProgressEvent<EventTarget>': lengthComputable, loaded, total

107       deregisterInteractionHandler: (evtType: EventType, handler: EventListener) => {
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  material_components_web/ripple/adapter.d.ts:41:5
    41     deregisterInteractionHandler<K extends EventType>(evtType: K, handler: SpecificEventListener<K>): void;
```

(and similarly so for `registerInteractionHandler`)

Comment (asyncLiz) [reply packages/mdc-switch/component.ts]: Our tsconfig [has that option set to `true`](https://github.com/material-components/material-components-web/blob/master/tsconfig-base.json#L14), not `false`. If we need our code to pass strict generic checks, wouldn't it make more sense to flip that option and then fix the type errors?

The best way to fix this instance would be to correct the type signature:
```ts
      deregisterInteractionHandler: <K extends EventType>(evtType: K, handler: SpecificEventListener<K>) => {
        this.nativeControl_.removeEventListener(evtType, handler, applyPassive());
      },
```

There are more places than mdc-switch that fail strict generic checks.

**Comment by mprobst** [comment]
@asyncLiz ack.

How this change came to be is that I fixed this internally in google's code repository (where this project's tsconfig does not apply, and we just have one shared compiler configuration for all projects). @abhiomkar is exporting my change.

So maybe flipping that bit could be a follow up? Not sure what your policy is here.

**Comment by asyncLiz** [comment]
Our tsconfig should probably match our internal compiler settings. However I think a bigger point I'm making is that we should probably fix the types and not mask the error with `as any`, which removes the type safety.

@abhiomkar what do you think? I can take on that work if you'd like.

**Comment by abhiomkar** [comment]
I think the origin of compiler error triggered internally when strict flag is enabled, is that correct @mprobst? 

I think we should first fix this internally. Are there any alternatives for fixing the type without using `as any`. Let's continue discussion at cl/290732368

Thanks!

**Comment by asyncLiz** [comment]
Closing in favor of #5511

**Pull request closed without merging**